src/utils/dev.ts

```diff
@@ -244,14 +244,15 @@ class NuxtDevServer extends EventEmitter {
       this.listener.server.on(
         'upgrade',
         async (req: any, socket: any, head: any) => {
-          if (
-            req.url.startsWith(
-              this._currentNuxt?.options.app.buildAssetsDir /* /_nuxt/ */,
-            )
-          ) {
+          const nuxt = this._currentNuxt
+          if (!nuxt) return
+          const viteHmrPath = (
+            nuxt.options.app.baseURL + nuxt.options.app.buildAssetsDir
+          ).replace(/\/\//g, '/')
```

Comment:
You might have better method to combine two URLs - feel free to updates

Reply:
Can we try ufo.joinURL?